tests/php/tests/includes/helper/test_class.wp-job-manager-com-auth-token.php

```diff
@@ -0,0 +1,263 @@
+
+		// Assert.
+		$this->assertIsString( $result );
+		$this->assertNotEmpty( get_user_meta( $user->ID, WP_Job_Manager_Com_Auth_Token::META_KEY ) );
+	}
+
+	public function testGenerate_WhenPassedPost_ShouldPersistMeta() {
+		// Arrange.
+		$instance = WP_Job_Manager_Com_Auth_Token::instance();
+		$post = $this->factory->post->create_and_get();
+
+		// Act.
+		$result = $instance->generate( 'post', $post->ID);
+
+		// Assert.
+		$this->assertIsString( $result );
+		$this->assertNotEmpty( get_post_meta( $post->ID, WP_Job_Manager_Com_Auth_Token::META_KEY ) );
+	}
+
+	public function testValidate_WhenPassedInvalidObjectType_ShouldReturnFalse() {
+		// Arrange.
+		$instance = WP_Job_Manager_Com_Auth_Token::instance();
+
+		// Act.
+		$result = $instance->validate( 'comment', 1, 'test' );
+
+		// Assert.
+		$this->assertFalse( $result );
+	}
+
+	public function testValidate_WhenPassedInvalidObjectID_ShouldReturnFalse() {
+		// Arrange.
+		$instance = WP_Job_Manager_Com_Auth_Token::instance();
+
+		// Act.
+		$result = $instance->validate( 'user', 'test', 'test' );
+
+		// Assert.
+		$this->assertFalse( $result );
+	}
+
+	public function testValidate_WhenPassedUserWithoutMeta_ShouldReturnFalse() {
+		// Arrange.
+		$instance = WP_Job_Manager_Com_Auth_Token::instance();
+		$user = $this->factory->user->create_and_get();
+
+		// Act.
+		$result = $instance->validate( 'user', $user->ID, 'test' );
+
+		// Assert.
+		$this->assertFalse( $result );
+	}
+
+	public function testValidate_WhenPassedUserWithMetaButInvalidToken_ShouldReturnFalse() {
+		// Arrange.
+		$instance = WP_Job_Manager_Com_Auth_Token::instance();
+		$user = $this->factory->user->create_and_get();
+		$token = $instance->generate('user', $user->ID);
+
+		// Act.
+		$result = $instance->validate( 'user', $user->ID, $token. 'a' );
+
+		// Assert.
+		$this->assertFalse( $result );
+	}
+
+	public function testValidate_WhenPassedPostWithMetaButInvalidToken_ShouldReturnFalse() {
+		// Arrange.
+		$instance = WP_Job_Manager_Com_Auth_Token::instance();
+		$post = $this->factory->post->create_and_get();
+		$token = $instance->generate('post', $post->ID);
+
+		// Act.
+		$result = $instance->validate( 'post', $post->ID, $token. 'a' );
+
+		// Assert.
+		$this->assertFalse( $result );
+	}
+
+	public function testValidate_WhenPassedUserWithMetaAndValidTokenTwice_ShouldReturnFalse() {
+		// Arrange.
+		$instance = WP_Job_Manager_Com_Auth_Token::instance();
+		$user = $this->factory->user->create_and_get();
+		$token = $instance->generate('user', $user->ID);
+
+		// Act.
+		$instance->validate( 'user', $user->ID, $token );
+		$result = $instance->validate( 'user', $user->ID, $token );
+
+		// Assert.
+		$this->assertFalse( $result );
+	}
+
+	public function testValidate_WhenPassedPostWithMetaAndValidTokenTwice_ShouldReturnFalse() {
+		// Arrange.
+		$instance = WP_Job_Manager_Com_Auth_Token::instance();
+		$post = $this->factory->post->create_and_get();
+		$token = $instance->generate('post', $post->ID);
```

Comment:
```suggestion
		$token = $instance->generate( 'post', $post->ID );
```

Reply:
Fixed in 3e75085c236778ceb0fe8e4f055aca2b6a1323e3. Thanks!